ecs-init/cache/cache_test.go

```diff
@@ -122,13 +123,18 @@ func TestDownloadAgentDownloadMD5Failure(t *testing.T) {
 
 	mockFS := NewMockfileSystem(mockCtrl)
 	mockGetter := NewMockhttpGetter(mockCtrl)
+	mockMetadata := NewMockinstanceMetadata(mockCtrl)
 
-	mockFS.EXPECT().MkdirAll(config.CacheDirectory(), os.ModeDir|0700)
-	mockGetter.EXPECT().Get(config.AgentRemoteTarballMD5()).Return(nil, errors.New("test error"))
+	gomock.InOrder(
+		mockFS.EXPECT().MkdirAll(config.CacheDirectory(), os.ModeDir|0700),
+		mockMetadata.EXPECT().Region().Return(config.DefaultRegionName, nil),
```

Comment:
(multiple occurrences) For the positive case of `Region()` returning a real value (as opposed to there being some error), please use a different return value than `config.DefaultRegionName`.  If you use `config.DefaultRegionName` it'll be more difficult to distinguish if the correct region is being used, since that's the fall-back when an error occurs.

Also: there needs to be a negative test case in here where `Region()` returns an error.  I don't see one.